iree/compiler/Conversion/LinalgToSPIRV/VectorizeMemref.cpp

```diff
@@ -0,0 +1,307 @@
+      ConversionPatternRewriter &rewriter) const override;
+};
+
+class ProcessTransferRead final
+    : public MemRefConversionPattern<vector::TransferReadOp> {
+ public:
+  using MemRefConversionPattern<
+      vector::TransferReadOp>::MemRefConversionPattern;
+  LogicalResult matchAndRewrite(
+      vector::TransferReadOp read, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    if (!memrefUsageAnalysis.transferConvert(read)) return failure();
+    vector::TransferReadOp::Adaptor adaptor(operands);
+    Value memref = adaptor.memref();
+    Location loc = read.getLoc();
+    unsigned ratio =
+        getBitWidth(memref.getType().cast<MemRefType>().getElementType()) /
+        getBitWidth(read.getMemRefType().getElementType());
+    SmallVector<Value, 4> indices(adaptor.indices().begin(),
+                                  adaptor.indices().end());
+    indices.back() = rewriter.create<SignedDivIOp>(
+        loc, indices.back(), rewriter.create<ConstantIndexOp>(loc, ratio));
+    if (getBitWidth(memref.getType().cast<MemRefType>().getElementType()) ==
+        getBitWidth(read.getVectorType())) {
+      Type elType = memref.getType().cast<MemRefType>().getElementType();
+      Value newLoad = rewriter.create<LoadOp>(loc, elType, memref, indices);
+      Type serializedVecType =
+          VectorType::get(read.getVectorType().getNumElements(),
+                          read.getVectorType().getElementType());
+      newLoad =
+          rewriter.create<vector::BitCastOp>(loc, serializedVecType, newLoad);
+      newLoad = rewriter.create<vector::ShapeCastOp>(loc, read.getVectorType(),
+                                                     newLoad);
+      rewriter.replaceOp(read, newLoad);
+    } else {
+      Value newRead = rewriter.create<vector::TransferReadOp>(
+          loc, read.getVectorType(), memref, indices);
+      rewriter.replaceOp(read, newRead);
+    }
+    return success();
+  }
+};
+
+class ProcessTransferWrite final
+    : public MemRefConversionPattern<vector::TransferWriteOp> {
+ public:
+  using MemRefConversionPattern<
+      vector::TransferWriteOp>::MemRefConversionPattern;
+  LogicalResult matchAndRewrite(
+      vector::TransferWriteOp write, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    if (!memrefUsageAnalysis.transferConvert(write)) return failure();
+    vector::TransferWriteOp::Adaptor adaptor(operands);
+    Value memref = adaptor.memref();
+    Location loc = write.getLoc();
+    unsigned ratio =
+        getBitWidth(memref.getType().cast<MemRefType>().getElementType()) /
+        getBitWidth(write.getMemRefType().getElementType());
+    SmallVector<Value, 4> indices(adaptor.indices());
+    indices.back() = rewriter.create<SignedDivIOp>(
+        loc, indices.back(), rewriter.create<ConstantIndexOp>(loc, ratio));
+    if (getBitWidth(memref.getType().cast<MemRefType>().getElementType()) ==
```

Comment:
Same comment as above for the `vector.transfer_read` rewrite. Why do we need to convert this to a store here?